implementation.js

```diff
@@ -114,13 +117,10 @@ if ('Map' in global) {
 var usingIterator = function (items) {
 	if (has(items, iteratorSymbol)) {
 		return items[iteratorSymbol]();
-	} else if ('Set' in global && 'Map' in global && isCallable(items.entries) && isCallable(items.values)) {
-		if (isSet(items)) {
-			return items.values();
-		}
-		if (isMap(items)) {
-			return items.entries();
-		}
+	} else if (hasMap && isCallable(items.entries) && isMap(items)) {
+		return items.entries();
```

Comment:
It would be ideal to cache `Map.prototype.entries` and `Set.prototype.values` at the module level, and `.call` them here, rather than relying on those methods being available on the instance/prototype